src/main/java/jsr223/kubernetes/KubernetesScriptEngine.java

```diff
@@ -133,13 +129,27 @@ else if (k8sDeleteOnly) {
         return resultValue;
     }
 
+    private void initializeEngine() {
+        bindings.addBindingsAsEnvironmentVariables(context);
+        setScriptEngineBehaviorFromEnv();
+    }
+
     /**********************************************/
     /* Kubernetes script engine auxiliary methods */
     /**********************************************/
 
     private void writeKubernetesManifestFile(String k8s_manifest) {
         // Write k8s manifest to file
+        Map<String, String> environment = bindings.getEnvironment();
         try {
+            for (Map.Entry<String, String> variable : environment.entrySet()) {
+                if (k8s_manifest.contains("${" + variable.getKey() + "}")) {
```

Comment:
you can define a local variable to contain the pattern

Reply:
alternatively you can put in the replacement map directly the correct pattern

Reply:
Right, done.